feat(charts): Page releases so that we get more results

wmak opened the pull request:
- This also adds a param to get a much lighter version of releases that
  only has the minimum params for charting

wmak committed:
feat(charts): Page releases so that we get more results

- This also adds a param to get a much lighter version of releases that
  only has the minimum params for charting

wmak committed:
ref: Update graph as we get releases

- This way the chart will have releases as we get them instead of being
  empty until all the releases have arrived

markstory commented on src/sentry/api/endpoints/organization_releases.py:
```diff
@@ -126,6 +126,7 @@ def get(self, request, organization):
         with_health = request.GET.get("health") == "1"
         flatten = request.GET.get("flatten") == "1"
         sort = request.GET.get("sort") or "date"
+        chart_only = request.GET.get("chartOnly") == "1"
```
I was wondering if we shouldn't have a 'timeseries' endpoint for releases. That would let us reduce the amount of transformation we need to do in the frontend as well.

A separate endpoint could let us use a larger per-page sizes, let us fetch only the properties we need from the database, and by-pass ORM hydration.

wmak replied:
:+1: So totally agree on all the benefits of the separate endpoint, realized the old implementation was still hitting snuba, so this new endpoint is taking a fraction of the time, for magnitudes more data 🥳 

But not sure if there was a specific transformation you had in mind with when you mentioned this?
> reduce the amount of transformation we need to do in the frontend

wmak committed:
feat(releases): Adding a stats endpoint for releases

- This adds a ReleaseStats endpoint that's specifically for adding
  releases to our timeseries
- Move from using extra for coalesce to django's Coalesce

wmak committed:
ref: reverting a diff on the serializer
